static/to_collect/admin-map-widget.js

```diff
@@ -1,52 +1,72 @@
     // Map control to switch type when geometry type is unknown
-    const options = opt_options || {}
+    constructor(opt_options) {
+        const options = opt_options || {}
 
-    const element = document.createElement("div")
-    element.className =
-        "switch-type type-" + options.type + " ol-control ol-unselectable"
-    if (options.active) {
-        element.classList.add("type-active")
-    }
-
-    const self = this
-    const switchType = function (e) {
-        e.preventDefault()
-        if (options.widget.currentGeometryType !== self) {
-            options.widget.map.removeInteraction(options.widget.interactions.draw)
-            options.widget.interactions.draw = new ol.interaction.Draw({
-                features: options.widget.featureCollection,
-                type: options.type,
-            })
-            options.widget.map.addInteraction(options.widget.interactions.draw)
-            options.widget.currentGeometryType.element.classList.remove("type-active")
-            options.widget.currentGeometryType = self
+        const element = document.createElement("div")
+        element.className =
+            "switch-type type-" + options.type + " ol-control ol-unselectable"
+        if (options.active) {
             element.classList.add("type-active")
         }
-    }
 
-    element.addEventListener("click", switchType, false)
-    element.addEventListener("touchstart", switchType, false)
+        super({
+            element: element,
+            target: options.target,
+        })
+        const self = this
+        const switchType = function (e) {
+            e.preventDefault()
+            if (options.widget.currentGeometryType !== self) {
+                options.widget.map.removeInteraction(options.widget.interactions.draw)
+                options.widget.interactions.draw = new ol.interaction.Draw({
+                    features: options.widget.featureCollection,
+                    type: options.type,
+                })
+                options.widget.map.addInteraction(options.widget.interactions.draw)
+                options.widget.currentGeometryType.element.classList.remove(
+                    "type-active",
+                )
+                options.widget.currentGeometryType = self
+                element.classList.add("type-active")
+            }
+        }
 
-    ol.control.Control.call(this, {
-        element: element,
-    })
+        element.addEventListener("click", switchType, false)
+        element.addEventListener("touchstart", switchType, false)
+    }
 }
-ol.inherits(GeometryTypeControl, ol.control.Control)
 
 // TODO: allow deleting individual features (#8972)
-{
-    const jsonFormat = new ol.format.GeoJSON()
-
-    function MapWidget(options) {
+class MapWidget {
+    constructor(options) {
         this.map = null
         this.interactions = { draw: null, modify: null }
         this.typeChoices = false
         this.ready = false
 
+        // Added for SPPNAUT admin
```

Comment:
addition to get the red color

Reply:
Maybe we could have the definition of the style in `custom-openlayers.html` and pass it in the options dictionary?